src/test/java/org/opensearch/flowframework/workflow/ToolStepTests.java

```diff
@@ -39,6 +49,9 @@ public void setUp() throws Exception {
             "test-id",
             "test-node-id"
         );
+        inputDataWithConnectorId = new WorkflowData(Map.of(CONNECTOR_ID, mockedConnectorId), "test-id", "test-node-id");
+        inputDataWithModelId = new WorkflowData(Map.of(MODEL_ID, mockedModelId), "test-id", "test-node-id");
+        inputDataWithAgentId = new WorkflowData(Map.of(AGENT_ID, mockedAgentId), "test-id", "test-node-id");
```

Comment:
the last argument `test-node-id` should be unique for each of these; it's the workflow step id.  Your tests are working because you're just hardcoding the map, but ideally the connector param should be the `createConnectorNodeName` string you're using.

Reply:
Updated